standards/JWT-SVID.md

```diff
@@ -0,0 +1,112 @@
+
+JWT-SVIDs are standard JWT tokens with a handful of restrictions applied. JWTs have historically proven difficult to get right, gaining a reputation in the security community as a technology which is likely to introduce vulnerabilities in its deployments and implementations. JWT-SVID takes steps to mitigate these problems as much as possible without breaking compatibility with existing applications and libraries.
+
+## 2. JOSE Header
+Historically, complexity introduced by the cryptographic agility of the JOSE header has led to a series of vulnerabilities in popular JWT implementations. To avoid such pitfalls, this specification greatly restricts the allowances originally afforded. This section describes the permitted registered headers, as well as their values. Any header not described here, registered or private, MUST NOT be included in the JWT-SVID JOSE Header.
+
+Only JWS is supported, and all header values MUST reside in a JWS Protected Header.
+
+### 2.1. Algorithm
+The `alg` header MUST be set to `RS256`.
+
+Most vulnerabilities in JWT come from mixing and matching of cryptographic techniques, so JWT-SVID defines only a single technique as valid. By taking this approach, we aim to sidestep the pitfalls associated with cryptographic agility. In the event that the above-defined technique needs to be changed, it will be done by releasing a new version of JWT-SVID.
+
+The `RS256` algorithm was chosen for a number of reasons. First, it is a goal of this specification to remain compatible with existing JWT libraries and applications. This largely restricts options to the algorithms defined in [RFC 7518 Section 3.1][5], ruling out the use of more efficient algorithms like those based on the Ed25519 curve.
+
+Second, the `RS256` algorithm is very widely implemented and supported by practically all JWT libraries and consumers that are capable of asymmetric token signing and validation. Finally, the RSA algorithm is less likely to suffer from critical implementation flaws than those based on elliptic curves.
+
+### 2.2. Key ID
+The `kid` header is optional.
+
+### 2.3. Type
+The `typ` header is optional. If set, its value MUST be either `JWT` or `JOSE`.
+
+## 3. JWT Claims
+The JWT-SVID specification does not introduce any new claims, though it does set some restrictions on the registered claims defined by [RFC 7519][1]. Registered claims not described in this document, in addition to private claims, MAY be used as implementers see fit. It should be noted, however, that reliance on claims which are not defined here may impact interoperability, as the producing and consuming applications must independently agree. Implementers should exercise caution when introducing additional claims and carefully consider the impact on SVID interoperability, particularly in environments where the implementer does not control both the producer and the consumer.
+
```

Comment:
Ported comment by @gcallaghan from draft spec:
```OIDC recommends collision resistant names for custom claims https://openid.net/specs/openid-connect-core-1_0.html#AdditionalClaims, in practice implementers accomplish this by using a URI namespace https://auth0.com/docs/api-auth/tutorials/adoption/scope-custom-claims#custom-claims. This might be a worthwhile addition to the spec. Also, it might be nice to have guidance for some .well_known url for a consumer to advertise supported claims.```

Reply:
I think we should steer clear of defining claims beyond the "standard" ones described here. There's a myriad of use cases and capturing them all in a generic way will be hard. Additionally, I'm personally a little hesitant to include arbitrary claims/metadata on identity documents due to interoperability concerns (which is the reason that this snippet of text was added).

I added a validation reference appendix at the end to try and make it clear which claims must be present, and how to validate them... reviewing this section, I can see that it's kind of hard to make that jump. Additionally, all the claims described here are mandatory. Perhaps we can add some text to this section highlighting the fact that all claims described here are mandatory, and/or insert a direct reference to the appendix?

Reply:
Yeah, what I was trying to convey is I wouldn't want to exclude implementors from including their own custom claims. I would suggest that if custom claims are added they SHOULD/MUST be collision resistant by using a URI namespace.

Reply:
Ah got it... I will add clarifying text here